spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/BoxControllerHelperService.java

```diff
@@ -95,16 +96,19 @@ public JSONObject listAllBoxesTable(HttpSession session, JSONObject json) {
       JSONObject j = new JSONObject();
       JSONArray jsonArray = new JSONArray();
       for (Box box : requestManager.listAllBoxes()) {
-        jsonArray.add("['" +
-          TableUtils.hyperLinkify("/miso/box/" + box.getId(), box.getName(), true) + "','" +
-          TableUtils.hyperLinkify("/miso/box/" + box.getId(), box.getAlias()) + "','" +
-          box.getLocationBarcode() + "','" + 
-          (box.getSize().getRows() * box.getSize().getColumns() - box.getFree()) + "/" + 
-          box.getSize().getRows() * box.getSize().getColumns() + "','" +
-          box.getSize().getRows() + "x" + box.getSize().getColumns() + "','" +
-          (box.getIdentificationBarcode() != null ? box.getIdentificationBarcode() : "") + "','" +
-          box.getUse().getAlias() + "','" +
-          (box.getIdentificationBarcode() != null ? box.getIdentificationBarcode() : "") + "']");
+        int filledSpaces = box.getSize().getRows() * box.getSize().getColumns() - box.getFree();
+        int availableSpaces = box.getSize().getRows() * box.getSize().getColumns();
+        JSONArray inner = new JSONArray();
+        
+        inner.add(TableHelper.hyperLinkify("/miso/box/" + box.getId(), box.getName()));
+        inner.add(TableHelper.hyperLinkify("/miso/box/" + box.getId(), box.getAlias()));
+        inner.add(box.getLocationBarcode());
+        inner.add(filledSpaces + "/" + availableSpaces);
+        inner.add(box.getSize().getRows() + "x" + box.getSize().getColumns());
+        inner.add(box.getUse().getAlias());
+        inner.add(isStringEmptyOrNull(box.getIdentificationBarcode()) ? "" : box.getIdentificationBarcode());
+        
```

Comment:
Aha, the error is thrown because (Location) "Barcode" was requested but not given in the CHS; it should be given above BoxUse alias. IdentificationBarcode is a hidden field, but served so users can search by it. I'll add the (Location)Barcode value in the CHS.